File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i in range(len(nums)):
            a=target-nums[i]
            if target-nums[i] in dic:
                return [i, dic[a]]
            else:
                dic[nums[i]]=i

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> temp = new HashMap<Integer, Integer>();
        int [] res = new int[2];
        for (int i=0; i<nums.length;i++){
            if (temp.containsKey(target-nums[i])){
                res[0]=i;
                res[1]=temp.get(target-nums[i]);
            }
            else{
                temp.put(nums[i],i);
            }
        }
        return res;
        
    }
}